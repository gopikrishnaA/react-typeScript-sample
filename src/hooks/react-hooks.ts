import {useEffect as layoutEffect} from 'react';

export const useEffect = layoutEffect;