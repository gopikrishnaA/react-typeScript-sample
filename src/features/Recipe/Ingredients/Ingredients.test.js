import React from "react";
import { shallow } from "enzyme";
import renderer from 'react-test-renderer';
import configureStore from "redux-mock-store";
import * as ReactReduxHooks from "../../../hooks/react-redux";
import Ingredient from "./Ingredient";


import Ingredients from "./Ingredients";

describe("Ingredients", () => {
  let wrapper;
  let store;

  let props = {
    ingredients: [{
      id: 123,
      name: 'test',
      measures: {
        us: {
          amount: 1000,
          unitShort: '$'
        }
      }
    }]
  };

  beforeEach(() => {
    store = configureStore()({});

    jest
      .spyOn(ReactReduxHooks, "useDispatch")
      .mockImplementation(() => store.dispatch);

    wrapper = shallow(<Ingredients {...props} />);
  });

  describe("on start", () => {
    it("renders correctly", () => {
      const tree = renderer
        .create(wrapper.find(".recipe-ings"))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it("dispach addItems to store", () => {
      wrapper.find(Ingredient).props().clicked(props.ingredients[0]);
      const actions = store.getActions();
      const expectedResult = [{ type: "ADD_SHOP_ITEM", item: {
        id: 123,
        amount: 1000,
        unit: '$',
        name: 'test'
      } }]
      expect(actions).toEqual(expectedResult);
    })
  });
});
