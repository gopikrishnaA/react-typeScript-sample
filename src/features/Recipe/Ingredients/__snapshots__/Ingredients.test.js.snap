// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ingredients on start renders correctly 1`] = `
<div
  className="recipe-ings d-flex flex-wrap flex-row   "
>
  <div
    className="d-flex align-items-center p-1 m-1"
    onClick={[Function]}
  >
    <i
      className="fas fa-cart-arrow-down mr-2 d-inline-block"
      title="Add to Shop List"
    />
    1000
     
    $
     
    test
  </div>
</div>
`;
